weather_stats/weatherstatsapi/weather/openweather/client.py

```diff
@@ -0,0 +1,54 @@
+from types import TracebackType
+from typing import Optional, Type
+
+from aiohttp import ClientSession
+from pydantic import ValidationError
+
+from weather.openweather.exceptions import OpenWeatherError
+from weather.openweather.schemas import GetWeatherData, GetWeatherParams
+from weather.openweather.settings import OPEN_WEATHER_SETTINGS
+
+
+class OpenWeatherClient:
+
+    _session: ClientSession
+
+    async def __aenter__(self) -> "OpenWeatherClient":
+        self._session = ClientSession()
+        return self
+
+    async def __aexit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
+        exc: Optional[BaseException],
+        tb: Optional[TracebackType],
+    ) -> None:
+        await self._session.close()
+
+    async def head_status(self) -> int:
+        async with self._session.get(
```

Comment:
As there is no guard in case of this method (and other similar) is called outside context manager, ugly `AttributeError` will be rised. Its better to introduce `_get_session()` method to get cached instance or create new session just for single method call.

Reply:
Added `property` so error msg is more descriptive